awscli/formatter.py

```diff
@@ -52,7 +52,8 @@ def _flush_stream(self, stream):
 
 
 class FullyBufferedFormatter(Formatter):
-    def __call__(self, operation, response, stream=None):
+    def __call__(self, command_name, response,
+                 is_response_paginated=False, stream=None):
```

Comment:
I think I can clean up this interface more.  Instead of having to pass a response object and a flag saying if it's a paginated response, I could just `isinstance` the response to check if it's paginated.  Either that or do a duck type check for `build_full_result` and follow that code path if response has that object.

Thoughts?